DnD-Montreal/session-tome
# Scaffold basic backend structure

## Description
Closes #60 
Closes #42
Closes #44 
Closes #46 
Closes #47 

Adds all migrations,  models, and controllers, according to architecture specification documents (ERD, Class Diagram, Domain Model).

Registers routes for newly created resource controllers.

Additionally, adds tests covering all controllers, defines preliminary model relationships, and resource request validation.

Installs Blueprint, which was used to generate the aforementioned files.

Omits basic blade views, since this platform will instead be using InteriaJS + React. 

## How to test
1. Ensure that all endpoints work, via Postman or Insomnia
   1. Get example collection for insomnia here: [SessionTome_Insomnia.zip](https://github.com/DnD-Montreal/session-tome/files/7229072/SessionTome_Insomnia.zip)
2. run test suite locally by running `php artisan test --env=env`
